app/split_integer.py

```diff
@@ -2,6 +2,6 @@ def split_integer(value: int, number_of_parts: int) -> list:
     parts = []
     for parts_left in range(number_of_parts, 0, -1):
         next_number = value // parts_left
-        parts.append(value // parts_left)
+        parts.append(next_number)
```

Comment:
The current logic does not handle the case where the value is less than the number of parts by adding zeros. Consider adding logic to ensure that zeros are added when necessary.